drivers/sensor/max17262/max17262.c

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright 2020 Matija Tudan
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(max17262, CONFIG_SENSOR_LOG_LEVEL);
+
+#include "max17262.h"
+
+#define DT_DRV_COMPAT maxim_max17262
+
+/**
+ * @brief Read a register value
+ *
+ * Registers have an address and a 16-bit value
+ *
+ * @param priv Private data for the driver
+ * @param reg_addr Register address to read
+ * @param valp Place to put the value on success
+ * @return 0 if successful, or negative error code from I2C API
+ */
+static int max17262_reg_read(struct max17262_data *priv, int reg_addr,
+			     int16_t *valp)
+{
+	uint8_t i2c_data[2];
+	int rc;
+
+	rc = i2c_burst_read(priv->i2c, DT_INST_REG_ADDR(0), reg_addr,
```

Comment:
I completely agree with all of your's I2C related comments. I wasn't so happy about them as well. But, please note that I used max17055 as a reference for this driver, so maybe change that there as well.

Reply:
> please note that I used max17055 as a reference for this driver, so maybe change that there as well

Yes, not all existing in-tree drivers are multi-instance. But all new drivers must be multi-instance. If you would like to help with the effort to fix up the existing drivers, I'm sure it would be much appreciated.